Classes/Queue/RedisQueue.php

```diff
@@ -296,10 +295,21 @@ protected function connectClient()
     {
         $host = isset($this->clientOptions['host']) ? $this->clientOptions['host'] : '127.0.0.1';
         $port = isset($this->clientOptions['port']) ? $this->clientOptions['port'] : 6379;
+        $password = isset($this->clientOptions['password']) ? $this->clientOptions['password'] : '';
         $database = isset($this->clientOptions['database']) ? $this->clientOptions['database'] : 0;
         // The connection read timeout should be higher than the timeout for blocking operations!
         $timeout = isset($this->clientOptions['timeout']) ? $this->clientOptions['timeout'] : round($this->defaultTimeout * 1.5);
-        $connected = $this->client->connect($host, $port, $timeout) && $this->client->select($database);
+
+        $connected = $this->client->connect($host, $port, $timeout);
+        if ($connected) {
+            if ($password !== '') {
+                if (!$this->client->auth($password)) {
+                    throw new JobQueueException('Redis authentication failed.', 1536735535);
+                }
+            }
```

Comment:
Nitpick: This could be rewritten to the IMO slightly more readable:
```
if ($connected && $password !== '') {
    $authSuccess = $this->client->auth($password);
    if ($authSuccess !== true) {
        throw new JobQueueException('Redis authentication failed.', 1536735535);
    }
}
```

Reply:
Thank you. Better code style...